addons/ui/fnc_progressBar.sqf

```diff
@@ -0,0 +1,172 @@
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_progressBar
+
+Description:
+    Opens a progress bar. Closes the currently active progress bar.
+
+Parameters:
+    _title      - Title of the progress bar <STRING>
+    _condition  - Execute every frame. If reports false, close the progress bar <CODE>
+    _totalTime  - Time for the progress bar to complete
+    _onSuccess  - Script to execute if the progress bar completed <CODE>
+    _onFailure  - Script to execute if the progress bar was aborted prematurely <CODE>
+    _arguments  - Arguments passed to the scripts (optional, default: []) <ANY>
+    _allowClose - Allow ESC key to abort the progress bar (optional, default: true) <BOOLEAN>
+    _blockInput - true: block keyboard + mouse,
+        false: block mouse, but only if _allowClose is true (optional, default: true) <BOOLEAN>
+
+Arguments:
+    _this:
+        #0 - same as _arguments <ANY>
+        #1 - true: success, false: failure <BOOLEAN>
+        #2 - elapsed time, not more than _totalTime <NUMBER>
+        #3 - total time, same as _totalTime <NUMBER>
+
+Returns:
+    Nothing
+
+Examples:
+    (begin example)
+        ["progress bar", {true}, 5, {hint "done"}, {hint "aborted"}] call CBA_fnc_progressBar;
+    (end)
+
+Author:
+    commy2
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+
+if (!hasInterface) exitWith {};
+
+if (canSuspend) exitWith {
+    isNil (uiNamespace getVariable _fnc_scriptName);
```

Comment:
What's this bit all about?

Reply:
Run the function unscheduled every time.

Reply:
Same as https://github.com/CBATeam/CBA_A3/blob/master/addons/settings/fnc_init.sqf#L44-L47

Reply:
Maybe prefer fnc_directCall for clarity?

Reply:
Yeah I just hadn't see the `uiNamespace` stuff before, was curious as to how that worked